packages/site/src/components/App.js

```diff
@@ -60,33 +61,38 @@ const Root = createStyledComponent('div', styles.app, {
 const Nav = createStyledComponent(_Nav, styles.nav);
 const Main = createStyledComponent('main', styles.main);
 
-const getDefaultDemo = (demos: Object): string => Object.keys(demos)[0];
-
 export default function App({ className, demos }: Props) {
+  let routes;
+
   if (!Array.isArray(demos) && demos.slug) {
     return <ComponentDoc {...demos} />;
+  } else {
+    routes = getRoutes();
   }
 
   const siteDemos = Array.isArray(demos) ? createKeyMap(demos, 'slug') : demos;
 
   return (
-    <Root className={className}>
-      <Nav demos={siteDemos} />
-      <Main>
-        <Switch>
-          <Route
-            path="/components/:componentId"
-            render={route => {
-              const componentId = route.match.params.componentId;
-              // $FlowFixMe
-              const selectedDemo = siteDemos[componentId];
-              return <ComponentDoc {...selectedDemo} />;
-            }}
-          />
-          <Redirect from="/" to={`/components/${getDefaultDemo(siteDemos)}`} />
-        </Switch>
-        <Footer />
-      </Main>
-    </Root>
+    <div>
+      <Root className={className}>
+        <Nav demos={siteDemos} />
+        <Main>
+          <Switch>
```

Comment:
IMHO, extracting all of this router stuff starting from `<Switch>` into a separate `Router` component would really help to clean up this file and separate responsibilities.  The `getRoutes` stuff could all be done in there too.  It is what I was trying to get at before, but had a bit of difficulty clarifying.   It seems like it would work and be pretty easy.

e.g.

```js
<div>
  <Root>
    <Nav />
    <Main>
      <Router />
    </Main>
  </Root>
</div>
```